src/common/java/net/minecraftforge/gradle/common/util/runs/RunConfigGenerator.java

```diff
@@ -179,7 +179,9 @@ private static BiFunction<String, Supplier<String>, Supplier<String>> makeClassp
             String resolvedClasspath = getResolvedClasspath(classpath.getFiles());
             if (supplier == null) return resolvedClasspath;
             String oldCp = supplier.get();
-            return oldCp == null || oldCp.isEmpty() ? resolvedClasspath : String.join(File.pathSeparator, oldCp, resolvedClasspath);
+            if (Strings.isNullOrEmpty(oldCp)) return resolvedClasspath;
+            if (Strings.isNullOrEmpty(resolvedClasspath)) return oldCp;
+            return String.join(File.pathSeparator, oldCp, resolvedClasspath);
```

Comment:
How does this guarantee that an empty string is returned if both are null-or-empty? Doesn't look like it does.

Reply:
```java
// resolvedClasspath != null by definition of getResolvedClasspath.
if (Strings.isNullOrEmpty(oldCp)) return resolvedClasspath;
// At this point, oldCp != null, so this returns a non-null value.
if (Strings.isNullOrEmpty(resolvedClasspath)) return oldCp;
// Finally, oldCp != null && resolvedClasspath != null
return String.join(File.pathSeparator, oldCp, resolvedClasspath);
```

I can add a comment to clarify if that would be helpful?

Reply:
Ah, that makes sense. Perhaps just add a `@NotNull` annotation to getResolvedClasspath, that should be good enough for the future.